src/axom/primal/operators/winding_number.hpp

```diff
@@ -215,7 +248,64 @@ double winding_number(const Point<T, 2>& q,
                       double edge_tol = 1e-8,
                       double EPS = 1e-8)
 {
-  return detail::curve_winding_number_recursive(q, c, false, edge_tol, EPS);
+  const int ord = c.getOrder();
+  if(ord <= 0) return 0.0;
+
+  // Early return is possible for must points + curves
+  if(!c.boundingBox().contains(q))
+  {
+    return detail::linear_winding_number(q, c[0], c[ord], edge_tol);
+  }
+
+  // The first vertex of the polygon is the t=0 point of the curve
+  Polygon<T, 2> approximating_polygon(1);
+  approximating_polygon.addVertex(c[0]);
+
+  // Need to keep a running total of the GWN to account for
+  //  the winding number of coincident points
+  double gwn = 0.0;
+  bool isCoincident = false;
+  detail::construct_approximating_polygon(q,
+                                          c,
+                                          false,
+                                          edge_tol,
+                                          EPS,
+                                          approximating_polygon,
+                                          gwn,
+                                          isCoincident);
+
+  // The last vertex of the polygon is the t=1 point of the curve
+  approximating_polygon.addVertex(c[ord]);
+
+  // Compute the integer winding numberof the closed curve
+  bool isOnEdge = false;
+  double closed_curve_wn =
+    winding_number(q, approximating_polygon, isOnEdge, false, edge_tol);
+
+  // Compute the fractional value of the closed curve
+  int n = approximating_polygon.numVertices();
+  double closure_wn = detail::linear_winding_number(q,
```

Comment:
[nit] The code might be slightly easier to understand if these two variable were marked `const`:
```suggestion
  // Compute the fractional value of the closed curve
  const int n = approximating_polygon.numVertices();
  const double closure_wn = detail::linear_winding_number(q,
```